pycheribuild/projects/cross/cheri_microkit.py

```diff
@@ -0,0 +1,304 @@
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+#
+import subprocess
+import threading
+import typing
+from pathlib import Path
+
+from .crosscompileproject import (
+    CompilationTargets,
+    CrossCompileAutotoolsProject,
+    DefaultInstallDir,
+    GitRepository,
+)
+from ..build_qemu import BuildCheriAllianceQEMU
+from ...qemu_utils import QemuOptions
+from .opensbi import BuildAllianceOpenSBI
+
+
+class BuildCheriseL4(CrossCompileAutotoolsProject):
+    target = "cheri-sel4"
+    repository = GitRepository(
+        "https://github.com/CHERI-Alliance/CHERI-seL4.git",
+        default_branch="std-cheri-riscv-microkit",
+        force_branch=True,
+    )
+    supported_architectures = (
+        CompilationTargets.FREESTANDING_RISCV64_PURECAP,
+        CompilationTargets.FREESTANDING_RISCV64,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def configure(self):
+        pass
+
+    def needs_configure(self):
+        return False
+
+    def compile(self, **kwargs):
+        pass
+
+    def install(self, **kwargs):
+        pass
+
+    def process(self):
+        super().process()
+
+
+class BuildCheriMicrokit(CrossCompileAutotoolsProject):
+    repository = GitRepository(
+        "https://github.com/CHERI-Alliance/CHERI-Microkit.git",
+        default_branch="std-cheri-riscv",
+        force_branch=True,
+    )
+    target = "cheri-microkit"
+    release_version = "2.0.1-dev"
+    dependencies = (
+        "cheri-alliance-qemu",
+        "cheri-alliance-gdb-native",
+        "cheri-alliance-opensbi",
+        "cheri-sel4",
+    )
+    native_install_dir = DefaultInstallDir.CHERI_ALLIANCE_SDK
+    is_sdk_target = True
+    needs_sysroot = False
+    supported_architectures = (
+        CompilationTargets.FREESTANDING_RISCV64_PURECAP,
+        CompilationTargets.FREESTANDING_RISCV64,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def setup(self):
+        super().setup()
+
+    @classmethod
+    def setup_config_options(cls, **kwargs):
+        super().setup_config_options(**kwargs)
+        cls.configs: str = typing.cast(
```

Comment:
I'm confused by the need to typing-cast here. I think the `add_config_option` method returns a handle for the option, not really the base type?

Reply:
I'm still a bit unsure about this because my experience with the type checker is limited. @arichardson do you have an opinion on this?